### PR TITLE
[FLINK-17974][docs] Extend docker documentation

### PR DESCRIPTION
**Pull request by rmetzger**
## What is the purpose of the change

While testing the new docker images, I found that some details were missing on the documentation page.

### CODE REVIEW

File: docs/ops/deployment/docker.md [hunk 1]
```diff
@@ -215,17 +241,16 @@ The `flink-conf.yaml` file must have write permission so that the Docker entry p
...
+If you want to enable plugins provided with Flink (in the `opt/` directory of the Flink distribution), you can pass the environment variable `ENABLE_BUILT_IN_PLUGINS` when you run the Flink image.
+The `ENABLE_BUILT_IN_PLUGINS` should contain a list of plugin jar file names separated by `;`. A valid plugin name is for example `flink-s3-fs-hadoop-{{site.version}}.jar`
 
-    ```sh
+```sh
     docker run \
-        --env ENABLE_BUILT_IN_PLUGINS=flink-plugin1.jar;flink-plugin2.jar \
-        flink:{% if site.is_stable %}{{site.version}}-scala{{site.scala_version_suffix}}{% else %}latest{% endif %} <jobmanager|standalone-job|taskmanager>
-    ```
+    --env ENABLE_BUILT_IN_PLUGINS=flink-plugin1.jar;flink-plugin2.jar \
```
Comment (zentol) [inline docs/ops/deployment/docker.md:251]: indentation seems different to other sections

File: docs/ops/deployment/docker.md [hunk 2]
```diff
@@ -105,18 +126,23 @@ To make the **job artifacts available** locally in the container, you can
 the *Flink Master* and *TaskManagers*:
 
     ```sh
+    FLINK_PROPERTIES="jobmanager.rpc.address: jobmanager"
+    docker network create flink-network
+
     docker run \
         --mount type=bind,src=/host/path/to/job/artifacts1,target=/opt/flink/usrlib/artifacts1 \
         --mount type=bind,src=/host/path/to/job/artifacts2,target=/opt/flink/usrlib/artifacts2 \
+        --rm --env FLINK_PROPERTIES="${FLINK_PROPERTIES}" --name=jobmanager --network flink-network \
```
Comment (zentol) [inline docs/ops/deployment/docker.md:135]: I'd prefer having everything on separate lines

File: docs/ops/deployment/docker.md [hunk 3]
```diff
@@ -73,18 +73,39 @@ The next chapters describe how to start a single Flink Docker container for vari
...
-docker run flink:{% if site.is_stable %}{{site.version}}-scala{{site.scala_version_suffix}}{% else %}latest{% endif %} taskmanager
+docker run \
+    --rm \
+    --name=taskmanager \
+    --network flink-network \
+    --env FLINK_PROPERTIES="${FLINK_PROPERTIES}" \
+    flink:{% if site.is_stable %}{{site.version}}-scala{{site.scala_version_suffix}}{% else %}latest{% endif %} taskmanager
 ```
 
+You can now access the Flink Webfrontend on [localhost:8081](http://localhost:8081/#/overview) or submit jobs like this `./bin/flink run ./examples/streaming/TopSpeedWindowing.jar`.
```
Comment (zentol) [inline docs/ops/deployment/docker.md:105]: This is the wrong place for this information. A user jumping straight to the job cluster docs will not see it.

File: docs/ops/deployment/docker.md [hunk 4]
```diff
@@ -73,18 +73,39 @@ The next chapters describe how to start a single Flink Docker container for vari
...
+
+We recommend using [Docker Compose](docker.html#session-cluster-with-docker-compose) or [Docker Swarm](docker.html#session-cluster-with-docker-swarm) for deploying Flink as a Session Cluster to ease system configuration.
```
Comment (zentol) [inline docs/ops/deployment/docker.md:107]: same as above

**Comment by rmetzger** [comment]
Thank you for your feedback. All addressed.

File: docs/ops/deployment/docker.md [hunk 5]
```diff
@@ -70,21 +70,43 @@ and deploys *TaskManagers* on demand so that you do not have to do it manually.
 
 The next chapters describe how to start a single Flink Docker container for various purposes.
 
+Once you've started Flink on Docker, you can access the Flink Webfrontend on [localhost:8081](http://localhost:8081/#/overview) or submit jobs like this `./bin/flink run ./examples/streaming/TopSpeedWindowing.jar`.
```
Comment (zentol) [inline docs/ops/deployment/docker.md:73]: maybe remove the duplicate paragraphs under ´Flink with Docker Compose#Usage` and `Flink with Docker Swarm`

Comment (rmetzger) [reply docs/ops/deployment/docker.md:73]: I would leave it. I don't think people read the entire page: They just jump to the section they are interested in, thus they won't see this.

**Comment by rmetzger** [comment]
Thanks for the review. I updated the zh files as well!

**Pull request closed without merging**